zerver/models.py

```diff
@@ -999,8 +1009,28 @@ def clean(self) -> None:
                 params={"name": name},
             )
 
+        if self.render_format_string:
+            found_group_set = set()
+            for m in group_match_regex.finditer(self.render_format_string):
+                group_name = m.group("group_name")
+                found_group_set.add(group_name)
+
+            # Report patterns missing in linkifier pattern.
+            #
+            # NOTE: We do not need to report the missing patterns in Render format string
+            # as some users may or may not want to render every patterns found in linkifer pattern.
```

Comment:
I think most correctly, we should be changing the existing check to allow fields in the linkifier pattern that are not present in the linkifier but are present in render_format_string, i.e. be merging the two `found_group_set` values.

Reply:
To do that we need to change the current error message:
https://github.com/zulip/zulip/blob/8cd9fc7a7411b29cf1a007fa49cf1d543f77063c/zerver/models.py#L985-L987
to: `Group %(name)r is not present in linkifier pattern.` since in such case we can't tell that the missing `name` in linkifier pattern was present in `url_format_string` or `render_format_string`.